cpp/include/cudf/rolling.hpp

```diff
@@ -110,6 +123,14 @@ struct current_row {
  */
 using range_window_type = std::variant<unbounded, current_row, bounded_closed, bounded_open>;
 
+/**
+ * @brief A request for a rolling aggregation on a column.
+ */
+struct rolling_request {
+  column_view values;                                ///< Elements to aggregate
+  std::unique_ptr<rolling_aggregation> aggregation;  ///< Desired aggregation
+};
```

Comment:
Let us paint the colour of this bikeshed.

I suspect perhaps it makes more sense to have the same interface as the `groupby` APIs. That is, accepting a:

```c++
struct rolling_request {
  column_view values; // elements to aggregate
  std::vector<std::unique_ptr<rolling_aggregation>> aggregations; // desired aggregations
}
```

and then the return value of `grouped_range_rolling_window` would be a `std::vector<std::unique_ptr<table>>` (`groupby.aggregate` returns the keys as a table and the results as a `std::vector<aggregation_result>` but an aggregation result is just a `std::vector<std::unique_ptr<column>>` so it feels like a `unique_ptr<table>` is just as good here).